xcode/Subconscious/Shared/Library/NaturalLanguage/TraceryDisambiguator.swift

```diff
@@ -12,57 +12,106 @@ struct TraceryContext: Hashable {
     var grammar: Grammar = [:]
 }
 
+enum DisambiguatorTag {
+    case journal
+    case project
+    case list
+    case link
+    case quote
+    case heading
+}
+
+struct DisambiguatorScore {
+    let tag: DisambiguatorTag
+    let weight: CGFloat
+}
+
+typealias DisambiguatorMatch = [DisambiguatorScore]
+
+extension DisambiguatorMatch {
+    func consolidate() -> DisambiguatorMatch {
+        // iterate over all entries and combine the scores of matching tags
+        var consolidated: DisambiguatorMatch = []
+        for score in self {
+            if let index = consolidated.firstIndex(where: { $0.tag == score.tag }) {
+                consolidated[index] = DisambiguatorScore(
+                    tag: score.tag,
+                    weight: consolidated[index].weight + score.weight
+                )
+            } else {
+                consolidated.append(score)
+            }
+        }
+        
+        return consolidated
+    }
+}
+
 protocol TraceryDisambiguatorRoute {
-    func match(_ input: String) async -> TraceryContext?
+    func classify(_ input: String) async -> DisambiguatorMatch
+}
+
+struct SubtextRoute: TraceryDisambiguatorRoute {
```

Comment:
I think we may want two sets of routes:

- Classification routes: takes input, passes it it to a succession of routes (up to max), and generate an array of DisambiguatorScores
- Result routes: takes input and generated DisambiguatorScores and generates a string result (may use Tracery or something else to do so)